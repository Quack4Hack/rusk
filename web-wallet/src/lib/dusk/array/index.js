export { default as arraysEqual } from "./arraysEqual";
export { default as shuffleArray } from "./shuffleArray";
export { default as findFirstNMatches } from "./findFirstNMatches";
