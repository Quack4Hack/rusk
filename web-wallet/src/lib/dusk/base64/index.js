export { default as base64ToUTF8 } from "./base64ToUTF8";
export { default as base64ToBytes } from "./base64ToBytes";
export { default as bytesToBase64 } from "./bytesToBase64";
export { default as utf8ToBase64 } from "./utf8ToBase64";
