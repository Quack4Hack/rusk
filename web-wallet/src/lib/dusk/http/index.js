export { default as failureToRejection } from "./failureToRejection";
