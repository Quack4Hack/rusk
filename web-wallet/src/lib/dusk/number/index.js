export { default as createNumberFormatter } from "./createNumberFormatter";
export { default as getDecimalSeparator } from "./getDecimalSeparator";
