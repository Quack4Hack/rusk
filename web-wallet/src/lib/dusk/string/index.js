export { default as calculateAdaptiveCharCount } from "./calculateAdaptiveCharCount";
export { default as hexStringToBytes } from "./hexStringToBytes";
export { default as makeClassName } from "./makeClassName";
export { default as middleEllipsis } from "./middleEllipsis";
