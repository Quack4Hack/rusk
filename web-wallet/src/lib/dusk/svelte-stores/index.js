export { default as createDataStore } from "./createDataStore";
