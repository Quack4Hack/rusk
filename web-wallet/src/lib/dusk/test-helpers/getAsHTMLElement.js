/**
 * Utility function to force the type checker to see
 * the return value of `querySelector` as a HTMLElement.
 *
 * @param {HTMLElement} container
 * @param {String} selector
 * @returns {HTMLElement}
 */
function getAsHTMLElement (container, selector) {
	// eslint-disable-next-line no-extra-parens
	return /** @type {HTMLElement} */ (container.querySelector(selector));
}

export default getAsHTMLElement;
