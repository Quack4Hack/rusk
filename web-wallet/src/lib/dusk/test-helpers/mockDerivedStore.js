import { get, writable } from "svelte/store";

/**
 * @param {*} initialValue
 * @param {*} fn
 */
function mockDerivedStore (initialValue, fn) {
	const store = writable(initialValue);
	const { set, subscribe } = store;
	const getMockedStoreValue = () => get(store);

	/** @param {*} value */
	const setMockedStoreValue = value => set(value);

	const derivedMockedStoreValue = () => fn(initialValue);

	return {
		derivedMockedStoreValue,
		getMockedStoreValue,
		setMockedStoreValue,
		subscribe
	};
}

export default mockDerivedStore;
