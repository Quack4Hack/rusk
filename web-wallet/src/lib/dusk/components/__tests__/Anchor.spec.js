import { afterEach, describe, expect, it } from "vitest";
import { cleanup } from "@testing-library/svelte";

import { renderWithSimpleContent } from "$lib/dusk/test-helpers";

import { Anchor } from "..";

describe("Anchor", () => {
  const baseProps = {
    href: "https://example.com",
  };

  const baseOptions = {
    props: baseProps,
  };

  afterEach(cleanup);

  it("should render the Anchor component", () => {
    const { container } = renderWithSimpleContent(Anchor, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should pass additional class names and attributes to the rendered element", () => {
    const props = {
      ...baseProps,
      className: "foo bar",
      href: "https://dusk.network",
      title: "Some title",
    };
    const { container } = renderWithSimpleContent(Anchor, {
      ...baseOptions,
      props,
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
