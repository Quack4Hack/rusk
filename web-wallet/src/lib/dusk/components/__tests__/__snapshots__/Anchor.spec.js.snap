// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Anchor > should pass additional class names and attributes to the rendered element 1`] = `
<a
  class="dusk-anchor foo bar"
  href="https://dusk.network"
  title="Some title"
>
  <span>
    some text
  </span>
</a>
`;

exports[`Anchor > should render the Anchor component 1`] = `
<a
  class="dusk-anchor"
  href="https://example.com"
>
  <span>
    some text
  </span>
</a>
`;
